src/ReflectionParameter.php

```diff
@@ -165,6 +165,11 @@ public function __toString()
      */
     public function allowsNull()
     {
+        // Allow builtin types to override
+        if ($this->parameterNode->getAttribute('prohibit_null', false)) {
```

Comment:
Does this belong to this PR or should a separate bugfix PR with connected unit tests?

Reply:
Yes, this is part of the handling for optional parameters without default values, which is new in this PR.

Reply:
OK. See above.